src/core_cli/Core_command.ml

```diff
@@ -206,14 +205,15 @@ let semgrep_core_with_one_pattern (config : Core_scan_config.t) : unit =
                      Parse_target.parse_and_resolve_name_warn_if_partial lang
                        file
                    in
+                   let path = Fpath.v file in
```

Comment:
see my comment above, if we use directly Fpath.t for files, we don't need to convert back again to Fpath.t here.

You can use !! also instead of Fpath.to_string if the path is used in format string %s

Reply:
Discussion on potential issues with replacing `Fpath.t` with `string` here at https://github.com/semgrep/semgrep/pull/9715/files/2852c5771fd5a447601064aa1757623d42d92bf7#r1489954087